src/ui/verificationSymbolStatusView.ts

```diff
@@ -0,0 +1,318 @@
+/* eslint-disable max-depth */
+import { commands, ExtensionContext, workspace, tests, Range, Position, Uri, TestRunRequest, TestController, TestRun, DocumentSymbol, TestItem, TestItemCollection, TextDocument, TestRunProfileKind, window } from 'vscode';
+import { Range as lspRange, Position as lspPosition } from 'vscode-languageclient';
+import { IVerificationSymbolStatusParams, PublishedVerificationStatus } from '../language/api/verificationSymbolStatusParams';
+import { DafnyLanguageClient } from '../language/dafnyLanguageClient';
+import CompilationStatusView from './compilationStatusView';
+import { Messages } from './messages';
+
+interface ResolveablePromise<T> {
+  promise: Promise<T>;
+  resolve(value: T): void;
+}
+
+interface ItemRunState {
+  run: TestRun;
+  startedRunningTime?: number;
```

Comment:
I think we had a consensus with @atomb @robin-aws @cpitclaudel and even @fabiomadge that displaying the running time for assertion batches is misleading, because there is already one method whose running time will be greater than all the other ones, for z3 to take the time to warm up. So we prefer resource unit.

However, here I think it's fine for now, because now the ExecutionEngine persists, this is for a top-level symbol, and the test API does not offer any other metric.
Just let's keep this in mind if later we want to extend this interface for single assertion batches.